File: jgd-rs/src/locales_keys.rs
pub enum LocalesKeys {
    En,
    FrFr,
    ItIt,
    JaJp,
    DeDe,
    PtBr,
    ArSa,
    CyGb,
}

impl From<LocalesKeys> for &str {
    fn from(value: LocalesKeys) -> Self {
        match value {
            LocalesKeys::En => "EN",
            LocalesKeys::FrFr => "FR_FR",
            LocalesKeys::ItIt => "IT_IT",
            LocalesKeys::JaJp => "JA_JP",
            LocalesKeys::DeDe => "DE_DE",
            LocalesKeys::PtBr => "PT_BR",
            LocalesKeys::ArSa => "AR_SA",
            LocalesKeys::CyGb => "CY_GB",
        }
    }
}

impl From<&str> for LocalesKeys {
    fn from(value: &str) -> Self {
        match value {
            "EN" => LocalesKeys::En,
            "FR_FR" => LocalesKeys::FrFr,
            "IT_IT" => LocalesKeys::ItIt,
            "JA_JP" => LocalesKeys::JaJp,
            "DE_DE" => LocalesKeys::DeDe,
            "PT_BR" => LocalesKeys::PtBr,
            "AR_SA" => LocalesKeys::ArSa,
            "CY_GB" => LocalesKeys::CyGb,
            _ => LocalesKeys::En,
        }
    }
}

File: jgd.js/src/locales-keys.ts
/**
 * Supported locale codes.
 */

export type LocaleCode =
  | "EN"
  | "FR_FR"
  | "IT_IT"
  | "JA_JP"
  | "DE_DE"
  | "PT_BR"
  | "AR_SA"
  | "CY_GB";

const SUPPORTED_LOCALES: LocaleCode[] = [
  "EN",
  "FR_FR",
  "IT_IT",
  "JA_JP",
  "DE_DE",
  "PT_BR",
  "AR_SA",
  "CY_GB",
];

/**
 * Convert string to valid locale code.
 */
export function localeFromString(value: string): LocaleCode {
  // Check if the value is a valid locale code
  if (SUPPORTED_LOCALES.includes(value as LocaleCode)) {
    return value as LocaleCode;
  }
  // Default to EN if not found
  return "EN";
}

/**
 * Get all supported locale codes.
 */
export function getSupportedLocales(): readonly LocaleCode[] {
  return SUPPORTED_LOCALES;
}

/**
 * Check if a locale code is supported.
 */
export function isLocaleSupported(locale: string): boolean {
  return SUPPORTED_LOCALES.includes(locale as LocaleCode);
}
